graph/db/graph.go

```diff
@@ -1,26 +1,93 @@
 package graphdb
 
-import "github.com/lightningnetwork/lnd/kvdb"
+import (
+	"time"
+
+	"github.com/lightningnetwork/lnd/graph/db/models"
+	"github.com/lightningnetwork/lnd/kvdb"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing/route"
+)
+
+// Config is a struct that holds all the necessary dependencies for a
+// ChannelGraph.
+type Config struct {
+	// KVDB is the kvdb.Backend that will be used for initializing the
+	// KVStore CRUD layer.
+	KVDB kvdb.Backend
+
+	// KVStoreOpts is a list of functional options that will be used when
+	// initializing the KVStore.
+	KVStoreOpts []KVStoreOptionModifier
+}
 
 // ChannelGraph is a layer above the graph's CRUD layer.
 //
 // NOTE: currently, this is purely a pass-through layer directly to the backing
 // KVStore. Upcoming commits will move the graph cache out of the KVStore and
 // into this layer so that the KVStore is only responsible for CRUD operations.
 type ChannelGraph struct {
+	graphCache *GraphCache
+
 	*KVStore
 }
 
 // NewChannelGraph creates a new ChannelGraph instance with the given backend.
-func NewChannelGraph(db kvdb.Backend, options ...OptionModifier) (*ChannelGraph,
+func NewChannelGraph(cfg *Config, options ...ChanGraphOption) (*ChannelGraph,
 	error) {
 
-	store, err := NewKVStore(db, options...)
+	opts := defaultChanGraphOptions()
+	for _, o := range options {
+		o(opts)
+	}
+
+	store, err := NewKVStore(cfg.KVDB, cfg.KVStoreOpts...)
```

Comment:
it looks like `NewKVStore` can just take the `Config`, maybe in another PR.

Reply:
see above comment about Config struct being ChannelGraph config and not KVStore config. 

We could add a KVStore config struct and then have the COnfig struct hold that struct but i think we can also after this series unexport anything KVStore related anyways so then would probs not make sense to have an exported KVStore config struct